control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -3480,3 +3491,237 @@ func healthCheckIdentityProvider(ctx context.Context, hcp *hyperv1.HostedControl
 	}
 	meta.SetStatusCondition(&hcp.Status.Conditions, condition)
 }
+
+func (r *HostedControlPlaneReconciler) reconcileDefaultSecurityGroup(ctx context.Context, hcp *hyperv1.HostedControlPlane) error {
+	logger := ctrl.LoggerFrom(ctx)
+	if hcp.Spec.Platform.Type != hyperv1.AWSPlatform {
+		// Not AWS platform, skip
+		return nil
+	}
+	if hcp.Status.Platform != nil && hcp.Status.Platform.AWS != nil && hcp.Status.Platform.AWS.DefaultWorkerSecurityGroupID != "" {
+		// Security group has already been created, nothing to do
+		return nil
+	}
+	validProvider := meta.FindStatusCondition(hcp.Status.Conditions, string(hyperv1.ValidAWSIdentityProvider))
+	if validProvider == nil || validProvider.Status != metav1.ConditionTrue {
+		logger.Info("Identity provider not ready. Skipping security group creation.")
+		return nil
+	}
+
+	originalHCP := hcp.DeepCopy()
+	var condition *metav1.Condition
+	sgID, creationErr := r.createAWSSecurityGroup(ctx, r.ec2Client, hcp.Spec.InfraID, hcp.Spec.Platform.AWS.CloudProviderConfig.VPC)
+	if creationErr != nil {
+		condition = &metav1.Condition{
+			Type:    string(hyperv1.AWSDefaultSecurityGroupCreated),
+			Status:  metav1.ConditionFalse,
+			Message: creationErr.Error(),
+			Reason:  hyperv1.AWSErrorReason,
+		}
+	} else {
+		condition = &metav1.Condition{
+			Type:    string(hyperv1.AWSDefaultSecurityGroupCreated),
+			Status:  metav1.ConditionTrue,
+			Message: hyperv1.AllIsWellMessage,
+			Reason:  hyperv1.AsExpectedReason,
+		}
+		hcp.Status.Platform = &hyperv1.PlatformStatus{
+			AWS: &hyperv1.AWSPlatformStatus{
+				DefaultWorkerSecurityGroupID: sgID,
+			},
+		}
+	}
+	meta.SetStatusCondition(&hcp.Status.Conditions, *condition)
+
+	if err := r.Client.Status().Patch(ctx, hcp, client.MergeFromWithOptions(originalHCP, client.MergeFromWithOptimisticLock{})); err != nil {
+		return fmt.Errorf("failed to update status: %w", err)
+	}
+
+	return creationErr
+}
+
+func awsSecurityGroupFilters(infraID string) []*ec2.Filter {
+	return []*ec2.Filter{
+		{
+			Name:   awssdk.String(fmt.Sprintf("tag:kubernetes.io/cluster/%s", infraID)),
```

Comment:
nit: to follow up, there's a awsClusterCloudProviderTagKey func we should export and reuse across the base code